src/System.Runtime.Extensions/src/System/Environment.Win32.cs

```diff
@@ -341,13 +342,17 @@ public static string UserName
                 }
 
                 return string.Empty;
+#else
+                return "Windows User";
+#endif
```

Comment:
It just requires yet another file, figuring out how to differentiate those files by name, etc. I don't have a strong preference either way.